internal/events/submanager.go

```diff
@@ -86,14 +87,15 @@ type SubscriptionManagerConf struct {
 }
 
 type subscriptionMGR struct {
-	conf          *SubscriptionManagerConf
-	db            kvstore.KVStore
-	rpc           eth.RPCClient
-	subscriptions map[string]*subscription
-	streams       map[string]*eventStream
-	closed        bool
-	cr            contractregistry.ContractResolver
-	wsChannels    ws.WebSocketChannels
+	conf               *SubscriptionManagerConf
+	db                 kvstore.KVStore
+	rpc                eth.RPCClient
+	subscriptions      map[string]*subscription
+	streams            map[string]*eventStream
+	closed             bool
+	cr                 contractregistry.ContractResolver
+	wsChannels         ws.WebSocketChannels
+	subscriptionsMutex sync.Mutex
```

Comment:
For concurrent access that is safe for both Reads & Writes, I suggest we use [RWMutex](https://pkg.go.dev/sync#RWMutex) and protect Read operations as well from concurrent access. Ex: https://github.com/hyperledger/firefly-ethconnect/blob/main/internal/events/submanager.go#L134